### PR TITLE
new keymap for projectkb alice

### PR DESCRIPTION
**Pull request by sidcarter**
A new keymap for the projectkb Alice 

## Description

Keymap optimized for my Mac. Changes or mods are:

* vim key bindings for arrows
* left and right command options
* reset moved from right to left
* added DFU_ARGS to rules.mk

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by yanfali** [comment]
Only orthogonally related, but could someone document the dip switch on the PCB?

**Comment by sidcarter** [comment]
oooh, yes. I agree. Are you ok for me to update the main readme doc, as part of this PR or would you want a different PR? let me know @yanfali

**Comment by yanfali** [comment]
Thank you @sidcarter

File: keyboards/projectkb/alice/keymaps/madhatter/rules.mk [hunk 1]
```diff
@@ -0,0 +1 @@
+DFU_ARGS = -d 0483:df11 -a 0 -s 0x08000000:leave
```
Comment (yanfali) [inline keyboards/projectkb/alice/keymaps/madhatter/rules.mk]: does this belong on top level rules.mk? @drashna ?

**Comment by sidcarter** [comment]
Those DFU_ARGS are actually pinned to the Alice pcb channel in project kb's discord.
So, I can actually move those to the main rules.mk. That specific line is strewn at random places in the QMK repo. Not sure, if that's how it's meant to be.

**Comment by yanfali** [comment]
> Those DFU_ARGS are actually pinned to the Alice pcb channel in project kb's discord.
> So, I can actually move those to the main rules.mk. That specific line is strewn at random places in the QMK repo. Not sure, if that's how it's meant to be.

Yeah, I think there's a PR to make it a standard value, but for now I think putting it in the main rules.mk would benefit the most people. Thanks

**Comment by sidcarter** [comment]
Changed, Updated and Tested. Still works like a charm. Thanks

File: keyboards/projectkb/alice/keymaps/madhatter/keymap.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
```
Comment (fauxpark) [inline keyboards/projectkb/alice/keymaps/madhatter/keymap.c]:
```suggestion
```

File: keyboards/projectkb/alice/keymaps/madhatter/keymap.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+enum layer_names {
+    _QWERTY,
+    _FNM
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
```
Comment (fauxpark) [inline keyboards/projectkb/alice/keymaps/madhatter/keymap.c]: This keycode isn't used anywhere.

File: keyboards/projectkb/alice/keymaps/madhatter/keymap.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  FNM
```
Comment (fauxpark) [inline keyboards/projectkb/alice/keymaps/madhatter/keymap.c]: And this keycode is already defined as `MO(_FNM)` below.

Comment (sidcarter) [reply keyboards/projectkb/alice/keymaps/madhatter/keymap.c]: removed.

File: keyboards/projectkb/alice/keymaps/madhatter/keymap.c (hunk 3 above)
Comment (sidcarter) [reply keyboards/projectkb/alice/keymaps/madhatter/keymap.c]: removed keycodes

File: keyboards/projectkb/alice/keymaps/madhatter/keymap.c (hunk 2 above)
Comment (sidcarter) [reply keyboards/projectkb/alice/keymaps/madhatter/keymap.c]: removed.

**Comment by sidcarter** [comment]
thanks @fauxpark - changes made and tested.

**Pull request merged**